crates/engine/tree/src/tree/mod.rs

```diff
@@ -1195,7 +1282,19 @@ where
     ///
     /// Assumes that `finish` has been called on the `persistence_state` at least once
     fn on_new_persisted_block(&mut self) {
-        self.state.tree_state.remove_before(self.persistence_state.last_persisted_block_number);
+        // TODO: is the sync target state the right thing to check for here?
+        let finalized = self.state.forkchoice_state_tracker.sync_target_state().and_then(|state| {
```

Comment:
we're actually missing a function to retrieve the last received valid FCU on the tracker.

lets add and add a helper for last_valid_finalized

Reply:
added helpers